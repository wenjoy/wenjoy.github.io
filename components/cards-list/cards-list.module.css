.container {
  display: flex;
  justify-content: start;
  padding: 5px;
  margin: 0;
  width: 100%;
  max-height: 300px;
  overflow: auto;
}

.wrap {
  flex-wrap: wrap;
}

.item-wrapper {
  margin: 0 15px 15px 0;
}
