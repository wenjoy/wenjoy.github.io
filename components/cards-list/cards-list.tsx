import cls from 'classnames'
import { Card } from '../card/card'
import styles from './cards-list.module.css'

interface CardsListProps {
  children?: (item: Card) => JSX.Element
  items: Card[]
  autoWrap?: boolean
}

export const CardsList: React.FC<CardsListProps> = ({
  items,
  children,
  autoWrap,
}) => {
  const ItemComp = children || ((item: Card) => <Card {...item} />)
  return (
    <div className={cls(styles.container, autoWrap && styles.wrap)}>
      {items.map((item) => (
        <div className={styles.itemWrapper} key={item.id}>
          <ItemComp {...item} />
        </div>
      ))}
    </div>
  )
}
