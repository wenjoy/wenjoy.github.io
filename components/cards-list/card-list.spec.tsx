import { render } from '@testing-library/react'
import { CardsList } from './cards-list'

describe('CardsList', () => {
  it('should render the cards list properly', () => {
    const { container } = render(
      <CardsList items={[{ id: '1', point: '5' }]} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render nothing if no items are provided', () => {
    const { container } = render(<CardsList items={[]} />)
    expect(container).toMatchSnapshot()
  })
})
