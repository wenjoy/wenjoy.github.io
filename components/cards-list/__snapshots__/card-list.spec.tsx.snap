// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList should render nothing if no items are provided 1`] = `
<div>
  <div
    class="container"
  />
</div>
`;

exports[`CardsList should render the cards list properly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="itemWrapper"
    >
      <div
        class="container small"
      >
        <div
          class="card frontface"
        >
           
          5
        </div>
        <div
          class="owner"
        />
      </div>
    </div>
  </div>
</div>
`;
